crates/router/src/core/payment_methods/cards.rs

```diff
@@ -271,6 +281,28 @@ pub async fn add_card_hs(
     ))
 }
 
+#[instrument(skip_all)]
+pub async fn decode_and_decrypt_locker_data(
+    key_store: &domain::MerchantKeyStore,
+    enc_card_data: String,
+) -> errors::CustomResult<Secret<String>, errors::VaultError> {
+    // Fetch key
+    let key = key_store.key.get_inner().peek();
+    // Decode
+    let decoded_bytes = hex::decode(&enc_card_data)
+        .into_report()
+        .change_context(errors::VaultError::ResponseDeserializationFailed)
+        .attach_printable("Failed to decode hex string into bytes")?;
+    // Decrypt
+    async { decrypt(Some(Encryption::new(decoded_bytes.into())), key).await }
```

Comment:
wrapping this with `async { ... }` not required, it's already async and can be directly awaited on.